### PR TITLE
prevent duplicated access requests creation from rems #431

### PR DESCRIPTION
**Pull request by neoflex**
When receiving a request to /api/rems we now check and ignore the items received if we already have active accesses created automatically for the same user, same dataset and same expiration date.

### CODE REVIEW

File: core/lcsb/rems.py [hunk 1]
```diff
@@ -155,37 +189,19 @@ def handle_rems_entitlement(data: Dict[str, str]) -> bool:
     return True
 
 
-def create_rems_entitlement(obj: Union[Access, User], 
-        application: str, 
-        dataset_id: str, 
-        user_oidc_id: str, 
-        email: str,
-        expiration_date: date) -> bool:
+def create_rems_entitlement(obj: Union[Access, User],
```
Comment (Fancien) [inline core/lcsb/rems.py:192]: This looks like a black-linted line. Should it not be all parameters on a new line?

File: core/lcsb/rems.py [hunk 2]
```diff
@@ -81,22 +102,36 @@ def handle_rems_entitlement(data: Dict[str, str]) -> bool:
...
-        expiration_date = date.today() + timedelta(
-            days=getattr(settings, 'ACCESS_DEFAULT_EXPIRATION_DAYS', 90)
-        )
+        expiration_date = build_default_expiration_date()
     else:
         expiration_date = isoparse(expiration_date).date()
 
-    logger.debug(f'REMS :: * data access request id: {application}, user_oidc_id: {user_oidc_id}, user_email: {email}, resource: {resource}')
+    logger.debug(
+        f'REMS :: * data access request id: {application}, user_oidc_id: {user_oidc_id}, user_email: {email}, resource: {resource}')
```
Comment (Fancien) [inline core/lcsb/rems.py:110]: This looks like a black-linted line. Ending parenthesis should be on a new line I think

File: core/lcsb/rems.py [hunk 3]
```diff
@@ -212,6 +228,26 @@ def create_rems_entitlement(obj: Union[Access, User],
     else:
         klass = obj.__class__.__name__
         raise TypeError(f'Wrong type of the object - should be Contact or User, is: {klass} instead')
-        
+
     new_logbook_entry.save()
     return True
+
+
+def build_access_notes_and_rems_user(application):
```
Comment (Fancien) [inline core/lcsb/rems.py:236]: Could you add some type hints to this method?

**Review by Fancien — CHANGES_REQUESTED** [review]
Looks good. 

Feel free to ignore the black-related comments, we will probably solve all of them at the end of the sprint as you mentioned some days ago.

File: core/lcsb/rems.py (hunk 2 above)
Comment (neoflex) [reply core/lcsb/rems.py]: I've reformatted the whole file with latest black

File: core/lcsb/rems.py (hunk 3 above)
Comment (neoflex) [reply core/lcsb/rems.py]: I've took advantage for that to split into two functions. Didn't really made sense to have the note and the user together.

**Review by Fancien — APPROVED** [review]
All good now. Thank you for the changes

**Pull request merged**